### DISCUSSION TITLE
npm private packages lookup from wrong (public) registry

### DISCUSSION THREAD

**Discussion by fjakop** [renovatebot/renovate#28905] [Request Help]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

GitLab, ghcr.io/renovatebot/renovate:37.338.0

### Please tell us more about your question or problem

A few months ago lookup of packages from private registry was ok. Now it seems that package-lock.json is no longer taken in account and all packages are tried to be looked up from https://registry.npmjs.org instead of the private one.

config.json hostrules

```
  "hostRules": [
    {
      "hostType": "npm",
      "matchHost": "repo.my.org",
      "username": "...",
      "password": "..."
    }
  ],
```

package-lock.json
```
    "node_modules/@my-org/ui-components": {
      "version": "1.7.0",
      "resolved": "https://repo.my.org/repository/npm/@my-org/ui-components/-/ui-components-1.7.0.tgz",
      "integrity": "sha512-T5bp...",
```
There's no .npmrc present.

Was there some major change to renovatebot which makes a change to config necessary?

Kind regards
Frank

### Logs (if relevant)

<details><summary>Logs</summary>

```
DEBUG: GET https://registry.npmjs.org/@my-org%2Fui-components = (code=ERR_NON_2XX_3XX_RESPONSE, statusCode=404 retryCount=0, duration=714) (repository=my-org/de.org.project)
DEBUG: Failed to look up npm package @my-org/ui-components (repository=my-org/de.org.project, packageFile=frontend/package.json, dependency=@my-org/ui-components)

```

</details>

**Comment by rarkins** [comment]
Renovate has never used the lockfile as the source of truth for "which registry should we look up?". Renovate will look for:
- `.npmrc` in the repo
- `npmrc` in Renovate config
- `packageRules.registryUrls` in Renovate config

If you don't have the file in the repo then you need one of the last two in your config.

Also, it's not even Renovate which writes to `package-lock.json` - that's done by calling `npm` itself. If `npm install` is run, and you don't have a `.npmrc` specifying an alternative registry (either in the repo, or Renovate writes it out to disk before calling `npm`), then any changed dependencies in the `package-lock.json` will come from the default npmjs registry.
